core/exec/src/mill/exec/GroupExecution.scala

```diff
@@ -483,7 +485,7 @@ trait GroupExecution {
   }
 
   def getValueHash(v: Val, task: Task[?], inputsHash: Int): Int = {
-    if (task.isInstanceOf[Task.Worker[?]]) inputsHash else v.##
+    if (task.isInstanceOf[Task.Worker[?]]) inputsHash else v.## + invalidateAllHashes
```

Comment:
This is hard to read. Does the `+ invalidateAllHashes` bind to both `if` branches?